doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-ESP8266/attach/esp8266-sample/user/user_main.c

```diff
@@ -0,0 +1,26 @@
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "uart.h"
+
+extern int main(void);
+    
+static void main_task(void *pvParameters)
+{
+    (void)pvParameters;
+    main();
+    for (;;);
+}
+
+void user_init(void)
+{
+    uart_init_new();
+    UART_SetBaudrate(UART0, 115200);
+    UART_SetPrintPort(UART0);
+    
+    portBASE_TYPE error = xTaskCreate(main_task, (const signed char *)"main_task",
```

Comment:
Do not need to cast task name:

```c

 BaseType_t xTaskCreate(    TaskFunction_t pvTaskCode,
                            const char * const pcName,
                            unsigned short usStackDepth,
                            void *pvParameters,
                            UBaseType_t uxPriority,
                            TaskHandle_t *pxCreatedTask
                          );
```

[Ref](http://www.freertos.org/a00125.html)